src/parser/WDL/parse.js

```diff
@@ -57,7 +59,375 @@ function logicParsingStage(ast, data) {
+    ? output.attributes.expression.name.toLowerCase()
+    : null;
+
+  if (valueType === Constants.IO_MEMBER_ACCESS) {
+    const index = callNames.indexOf(output.attributes.expression.attributes.lhs.source_string);
+
+    if (index > -1) {
+      output.attributes.expression.attributes.lhs.source_string = `${namespaces[index]}_${callNames[index]}`;
+    }
+  }
+
+  return output;
+}
+
+function getNamespacesAndCallNames(calls) {
+  const namespaces = [];
+  const callNames = [];
+
+  calls.forEach((c) => {
+    const arr = c.split(Constants.NS_SPLITTER);
+    namespaces.push(arr.shift());
+    callNames.push((arr).join(Constants.NS_SPLITTER));
+  });
+
+  return [namespaces, callNames];
+}
+
+/** Returns ast with renamed calls, calls inputs and workflow outputs (xxx.xxx -> xxx_xxx) */
+function updateFirstAst(firstAst, calls) {
+  const resAst = firstAst;
+  const [namespaces, callNames] = getNamespacesAndCallNames(calls);
+
+  resAst.attributes.body.list = resAst.attributes.body.list
+    .map((item) => {
+      if (item.name.toLowerCase() !== Constants.WORKFLOW) {
+        return item;
+      }
+
+      const change = list => list.map((i) => {
+        if (i.name.toLowerCase() !== Constants.DECLARATION) {
+          if (i.name.toLowerCase() === Constants.CALL) {
+            if (i.attributes.body && i.attributes.body.attributes && i.attributes.body.attributes.io) {
+              i.attributes.body.attributes.io.list = i.attributes.body.attributes.io.list
+                .map(io => proceedCallInputs(io, callNames, namespaces));
+            }
+
+            if (calls.includes(i.attributes.task.source_string)) {
+              i.attributes.task.source_string = i.attributes.task.source_string
+                .split(Constants.NS_SPLITTER).join(Constants.NS_CONNECTOR);
+
+              return i;
+            }
+
+            return i;
+          } else if (i.name.toLowerCase() === Constants.WF_OUTPUTS) {
+            i.attributes.outputs.list = i.attributes.outputs.list
+              .map(output => proceedWfOutput(output, namespaces, callNames));
+
+            return i;
+          }
+
+          i.attributes.body.list = change(i.attributes.body.list);
+
+          return i;
+        }
+
+        return i;
+      });
+
+      item.attributes.body.list = change(item.attributes.body.list);
+      return item;
+    });
+
+  return resAst;
+}
+
+function getPreparedSubWDLs(opts = {}) {
+  const { wdlArray, baseURI, imports } = opts;
+  const res = {};
+  let status = true;
+  let message = '';
+
+  if (wdlArray) {
+    wdlArray.forEach((item) => {
+      res[item.name] = item.wdl;
+    });
+  }
+
+  const promises = [];
+  /* eslint-disable */
+  _.forEach(imports.map(imp => imp.uri), (importUri) => {
+    if (!res[importUri]) {
+      if (importUri.indexOf('file://') === 0) {
+        message = '"file://" protocol for imports is not supported';
+        status = false;
+
+        // exit forEach
+        return false;
```

Comment:
Such comments might be a good sign of a high cognitive complexity of a method. Let's refactor it.

Reply:
Fixed